samza-kafka/src/test/scala/org/apache/samza/system/kafka/TestKafkaSystemAdmin.scala

```diff
@@ -59,32 +63,34 @@ object TestKafkaSystemAdmin extends KafkaServerTestHarness {
   var producer: KafkaProducer[Array[Byte], Array[Byte]] = null
   var metadataStore: TopicMetadataStore = null
   var producerConfig: KafkaProducerConfig = null
-  var brokers: String = null
+  var systemAdmin: KafkaSystemAdmin = null
 
-  def generateConfigs() = {
+  override def generateConfigs(): Seq[KafkaConfig] = {
     val props = TestUtils.createBrokerConfigs(numBrokers, zkConnect, true)
     props.map(KafkaConfig.fromProps)
   }
 
-  @BeforeClass
-  override def setUp {
-    super.setUp
+  @Before
```

Comment:
Does this significantly increase test times?

Reply:
The purpose of this change is not to make test faster. It is to make each test independent of each other. If we use BeforeClass and AfterClass tags, then these methods need to be static and the variables instantiated (i.e. producer, systemAdmin) needs to be static as well. These variables will then be instantiated exactly once for the test suite and used by all unit tests in this test suite. Thus kind allows interference between tests and make make test development difficult.

During development of this patch the interference must have caused some problem which required me to make this change. I don't exactly remember what that problem is now. But in general it seems like good practice to make unit test independent of each other. What do you think?

Reply:
These tests are more integration tests than unit tests since they bring up ZK and Kafka. This class has at least 8 tests, so wondering how much of a time difference starting ZK + Kafka 8x vs 1x makes. If it adds say ~ a minute to build times, I'd try to find a workaround.

Reply:
@prateekm Thanks. I have updated the patch to fix the issue.